docs/dev/table/connectors/datagen.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "DataGen SQL Connector"
+nav-title: DataGen
+nav-parent_id: sql-connectors
+nav-pos: 4
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+<span class="label label-primary">Scan Source: Bounded</span>
+<span class="label label-primary">Scan Source: UnBounded</span>
+
+* This will be replaced by the TOC
+{:toc}
+
+The Datagen connector allows for reading by data generation rules.
+
+The Datagen connector can work with [Computed Column syntax]({{ site.baseurl }}/dev/table/sql/create.html#create-table).
+This allows you to generate records flexibly.
+
+The Datagen connector is built-in.
+
+<span class="label label-danger">Attention</span> Not support complex types: Array, Map, Row. Please construct these types by computed column.
+
+How to create an Datagen table
+----------------
+
+For each field, there are two ways to generate data:
+
+- Random generator: default, you can specify random max and min values. For char/varchar/string, the length can be specified.
+- Sequence generator: you can specify sequence start and end values.
```

Comment:
explain more about the behavior of after reaching the end value ?

Reply:
`default` -> `By default`.